kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -177,6 +169,11 @@ class KinesisSink private (
         log.info(s"Writing ${batch.size} Thrift records to Kinesis stream $streamName.")
         writeBatchToKinesis(batch).onComplete {
           case Success(s) =>
+            // We only need to keep track of Kinesis health if no SQS buffer is configured.
+            // We also need a way to recover from an SQS outage without writing to SQS,
+            // as there's no guarantee the latter will happen.
+            if (buffer.isEmpty || sqsQueueExists(buffer.get.sqsClient, buffer.get.sqsBufferName))
```

Comment:
How about

```scala
if (buffer.isEmpty || (outage && sqsQueueExists(buffer.get.sqsClient, buffer.get.sqsBufferName)))
```

So we avoid the expensive `sqsQueueExists` call during normal operation when everything is healthy already.

Reply:
Done.